buzsaki_lab_to_nwb/petersen_code/petersenmiscdatainterface.py

```diff
@@ -0,0 +1,176 @@
+            trial_start_frames = trial_info['start'][0][0]
+            n_trials = len(trial_start_frames)
+            trial_end_frames = trial_info['end'][0][0]
+            trial_stat = trial_info['stat'][0][0]
+            trial_stat_labels = [x[0][0] for x in trial_info['labels'][0][0]]
+            cooling_info = trial_info['cooling'][0][0]
+            cooling_map = dict({0: "Cooling off", 1: "Pre-Cooling", 2: "Cooling on", 3: "Post-Cooling"})
+            trial_error = trial_info['error'][0][0]
+            error_trials = np.array([False]*n_trials)
+            error_trials[np.array(trial_error).astype(int)-1] = True  # -1 from Matlab indexing
+
+            trial_starts = []
+            trial_ends = []
+            trial_condition = []
+            for k in range(n_trials):
+                trial_starts.append(take_frame_to_time[trial_start_frames[k]])
+                trial_ends.append(take_frame_to_time[trial_end_frames[k]])
+                nwbfile.add_trial(start_time=trial_starts[k], stop_time=trial_ends[k])
+                trial_condition.append(trial_stat_labels[int(trial_stat[k])-1])
+
+            nwbfile.add_trial_column(
+                name='condition',
+                description="Whether the maze condition was left or right.",
+                data=trial_condition
+            )
+            nwbfile.add_trial_column(
+                name='error',
+                description="Whether the subject made a mistake.",
+                data=error_trials
+            )
+
+            if 'temperature' in trial_info:  # Some sessions don't have this for some reason
+                trial_temperature = trial_info['temperature'][0][0]
+                nwbfile.add_trial_column(
+                    name='temperature',
+                    description="Average brain temperature for the trial.",
+                    data=trial_temperature
+                )
+
+            if len(cooling_info) == n_trials:  # some sessions had incomplete cooling info
+                trial_cooling = []
+                for k in range(n_trials):
+                    trial_cooling.append(cooling_map[int(cooling_info[k])])
+                nwbfile.add_trial_column(
+                    name='cooling state',
+                    description="The labeled cooling state of the subject during the trial.",
+                    data=trial_cooling
+                )
+
+        # Position
+        animal_file_path = session_path / "animal.mat"
+
+        if animal_file_path.is_file():
+            behavioral_processing_module = check_module(nwbfile, 'behavior', "Contains processed behavioral data.")
+
+            animal_mat = loadmat(str(animal_file_path))['animal']
+            animal_time = animal_mat['time'][0][0][0]
+            animal_time_diff = np.diff(animal_time)
+            animal_time_kwargs = dict()
+            if all(animal_time_diff == animal_time_diff[0]):
```

Comment:
Something like [this](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/4f1280a7b86fe6f26ecd7d0daedb0594aa00dae7/nwbinspector/nwbinspector.py#L189) might be even better

Reply:
That is a good deal more sophisticated - would it be worth importing it from that repo and adding nwbinspector to the requirements or just copy/paste the function over?

Reply:
Can you copy/paste this into nwb conversion tools? I think it would be useful for many of our conversions

Reply:
Done, and updated here as well.